### PR TITLE
 RavenDB-9572 Provide a way to get revisions by change vectors

### PR DESCRIPTION
**Pull request by fitzchak**

### CODE REVIEW

File: src/Raven.Client/Documents/Session/DocumentSessionRevisions.cs [hunk 1]
```diff
@@ -18,14 +19,43 @@ public DocumentSessionRevisions(InMemoryDocumentSessionOperations session) : bas
         {
         }
 
-        public List<T> GetFor<T>(string id, int start = 0, int pageSize = 25)
+        public List<T> GetFor<T>(string id, int start = 0, int pageSize = 25, bool metadataOnly = false)
```
Comment (ayende) [inline src/Raven.Client/Documents/Session/DocumentSessionRevisions.cs:22]: What is the use of `metadataOnly` here? You aren't _getting_ the metadata here.

File: src/Raven.Client/Documents/Session/DocumentSessionRevisions.cs [hunk 2]
```diff
@@ -18,14 +19,43 @@ public DocumentSessionRevisions(InMemoryDocumentSessionOperations session) : bas
...
+            var operation = new GetRevisionOperation(Session, id, start, pageSize, metadataOnly);
 
             var command = operation.CreateRequest();
             RequestExecutor.Execute(command, Context, sessionInfo: SessionInfo);
             operation.SetResult(command.Result);
-            return operation.Complete<T>();
+            return operation.GetRevisionsFor<T>();
+        }
+
+        public List<MetadataAsDictionary> GetMetadataFor(string id, int start = 0, int pageSize = 25)
```
Comment (ayende) [inline src/Raven.Client/Documents/Session/DocumentSessionRevisions.cs:32]: What is the scenario that is enabled by allowing to get metadata of a range of revisions?

File: src/Raven.Client/Documents/Session/DocumentSessionRevisions.cs (hunk 1 above)
Comment (fitzchak) [reply src/Raven.Client/Documents/Session/DocumentSessionRevisions.cs:22]: This is a mistake, should have been removed. I'll fix that.

File: src/Raven.Client/Documents/Session/DocumentSessionRevisions.cs (hunk 2 above)
Comment (fitzchak) [reply src/Raven.Client/Documents/Session/DocumentSessionRevisions.cs:32]: To get their change vector and last modified date. This way you can load iterate and load just the revision that you want.

File: src/Raven.Client/Documents/Session/DocumentSessionRevisions.cs (hunk 1 above)
Comment (fitzchak) [reply src/Raven.Client/Documents/Session/DocumentSessionRevisions.cs:22]: https://github.com/ravendb/ravendb/pull/4538

**Pull request merged**